lib/Parser/StandardChars.cpp

```diff
@@ -261,6 +261,11 @@ END {
         set.SetNotRanges(setAllocator, numNewlinePairs, newlineStr);
     }
 
+    void StandardChars<char16>::SetFullSet(ArenaAllocator* setAllocator, CharSet<Char> &set)
+    {
+        set.SetRange(allocator, MinChar, MaxChar);
```

Comment:
>set.SetRange [](start = 8, length = 12)

This is fine as-is, but for memory usage reasons (and some runtime efficiency), I'd prefer if you could make the set the equivalent of `[^<none>]` instead of the equivalent of `[\x00-\u{1FFFFF}]` as you've done here. You should be able to use `SetNotRanges(0, <empty string>)` to accomplish this, but if there's some assertions or preconditions there that you're not using empty string, you might need a new method.

Reply:
SetNotRanges with 0 sorted pairs currently literally does what I've done here:
https://github.com/Microsoft/ChakraCore/blob/bffb8850cd491ba148b98c4ee9f322312d374c56/lib/Parser/CharSet.cpp#L1137-L1150

All the NotRanges seem to be handled in that way.

Reply:
Ah, that is surprising to me since we have a concept of a negated set which is more memory efficient. Possibly that was a later addition than the sets themselves. I'll look into this at a later time; won't block the PR on it.

Reply:
I'll change dotAll one to use SetNotRanges so if that method is updated to use negated sets this will benefit from the change.

Reply:
Sounds good!